checkasm/checkasm.c

```diff
@@ -494,18 +473,26 @@ static void check_cpu_flag(const char *const name, unsigned flag) {
     const unsigned old_cpu_flag = state.cpu_flag;
 
     flag |= old_cpu_flag;
-    dav1d_set_cpu_flags_mask(flag);
-    state.cpu_flag = dav1d_get_cpu_flags();
+    ass_set_cpu_flags_mask(flag);
+    state.cpu_flag = ass_get_cpu_flags();
 
     if (!flag || state.cpu_flag != old_cpu_flag) {
+        ASS_Renderer *renderer = ass_renderer_init(state.library);
```

Comment:
Is there a need to initialize a renderer for assembly functions? Looks like overkill for me. Without the renderer even the global cpu masking can be localized into this function instead.

Reply:
`ass_renderer_init` is responsible for selecting (or producing) a BitmapEngine struct.

Reply:
I think it's better to extract `BitmapEngine` selection from `ass_renderer_init()` into another function. In that case it's possible to make `flag_mask` into parameter of that function and eliminate global variable and access function altogether.